source/opt/instrument_pass.cpp

```diff
@@ -331,23 +380,40 @@ uint32_t InstrumentPass::GetOutputBufferId() {
                                desc_set_);
     deco_mgr->AddDecorationVal(output_buffer_id_, SpvDecorationBinding,
                                GetOutputBufferBinding());
-    // Look for storage buffer extension. If none, create one.
-    if (!get_feature_mgr()->HasExtension(
-            kSPV_KHR_storage_buffer_storage_class)) {
-      const std::string ext_name("SPV_KHR_storage_buffer_storage_class");
-      const auto num_chars = ext_name.size();
-      // Compute num words, accommodate the terminating null character.
-      const auto num_words = (num_chars + 1 + 3) / 4;
-      std::vector<uint32_t> ext_words(num_words, 0u);
-      std::memcpy(ext_words.data(), ext_name.data(), num_chars);
-      context()->AddExtension(std::unique_ptr<Instruction>(
-          new Instruction(context(), SpvOpExtension, 0u, 0u,
-                          {{SPV_OPERAND_TYPE_LITERAL_STRING, ext_words}})));
-    }
+    AddStorageBufferExt();
   }
   return output_buffer_id_;
 }
 
+uint32_t InstrumentPass::GetInputBufferId() {
+  if (input_buffer_id_ == 0) {
+    // If not created yet, create one
+    analysis::DecorationManager* deco_mgr = get_decoration_mgr();
+    analysis::TypeManager* type_mgr = context()->get_type_mgr();
+    analysis::Type* reg_uint_rarr_ty =
+        GetUintRuntimeArrayType(deco_mgr, type_mgr);
+    analysis::Struct ibuf_ty({reg_uint_rarr_ty});
+    analysis::Type* reg_ibuf_ty = type_mgr->GetRegisteredType(&ibuf_ty);
+    uint32_t ibufTyId = type_mgr->GetTypeInstruction(reg_ibuf_ty);
+    deco_mgr->AddDecoration(ibufTyId, SpvDecorationBlock);
+    deco_mgr->AddMemberDecoration(ibufTyId, 0, SpvDecorationOffset, 0);
```

Comment:
I think the same issue can come up here.  You may get an existing type used for something else.  Then adding the decorations could cause a decoration to be applied to something to which it should not be applied.

Reply:
I believe my logic from the previous comment also applies to this type. Any pre-existing struct type with a runtime array will have to be a block and thus will have to be decorated with at least one Offset. So I will not get a pre-existing type here so I will be free to decorate it.

Reply:
Okay.  Again document it, and reference the spec for any assumption you make.